### PR TITLE
Implement training time rtc for pi0, pi0.5 and smolvla

### PR DESCRIPTION
**Pull request by pkooij**
## Title

Add training-time RTC for pi0/pi05/smolvla

## Type / Scope

- **Type**: Feature
- **Scope**: policies/rtc, pi0, pi05, smolvla, docs

## Summary / Motivation

This adds training-time RTC (action-prefix conditioning) to LeRobot. 

The change introduces a small shared helper for delay sampling and prefix/postfix masking, and integrates this into PI0/PI05 and SmolVLA training loss. This mirrors the method in the paper and keeps inference unchanged while enabling training time RTC.

## Related issues

- Fixes / Closes: none
- Related: none

## What changed

- Added training-time RTC config and helper utilities (delay distribution, prefix/postfix masks).
- Implemented training‑time RTC in PI0/PI05/SmolVLA losses
- Added a training‑time RTC doc 
- No breaking changes; new config is optional.

## How was this tested (or how to run locally)

- Tests added: list new tests or test files.
- Manual checks / dataset runs performed.
- Instructions for the reviewer

Example:

- Ran the relevant tests:

  ```bash
  pytest -q tests/policies/rtc/test_training_time_rtc.py
  ```

- Reproduce with a quick example or CLI:

  ```bash
  lerobot-train \
    --policy.type=pi0 \
    --dataset.repo_id=your/dataset \
    --policy.rtc_training_config.enabled=true \
    --policy.rtc_training_config.min_delay=0 \
    --policy.rtc_training_config.max_delay=6 \
    --policy.rtc_training_config.delay_distribution=UNIFORM
  ```

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [x] All tests pass locally (`pytest`)
- [x] Documentation updated
- [ ] CI is green

## Reviewer notes

- Anyone in the community is free to review the PR.

### CODE REVIEW

**Comment by helper2424** [comment]
Nice, you have started it early than me - 🔥 Lets check :)

**Comment by Lakesenberg** [comment]
let's go

**Comment by xianglunkai** [comment]
@pkooij 
looks have some bugs for training with pi05?

**Comment by xianglunkai** [comment]
@pkooij 
any update?
